tests/models/test_model.py

```diff
@@ -549,90 +547,107 @@ def test_gridmodel_setup(tmpdir):
-#         geoda.vector.geometry.to_crs(3857).buffer(0.1).to_crs(crs)
-#     )
-#     with pytest.raises(ValueError, match="Geometry of data and vector do not match"):
-#         vector_model.set_vector(data=geoda_test)
-#     param3 = vector_model.vector["param1"].sel(index=slice(0, 3)).drop_vars("geometry")
-#     with pytest.raises(ValueError, match="Index coordinate of data variable"):
-#         vector_model.set_vector(data=param3, name="param3")
-#     vector_model.set_vector(data=geoda, overwrite_geom=True, name="zs")
-#     assert "param1" not in vector_model.vector
-#     assert "zs" in vector_model.vector
-
-#     # test write vector
-#     vector_model.set_vector(data=gdf)
-#     vector_model.root.set(str(tmpdir), mode="w")
-#     # netcdf+geojson --> tested in test_vectormodel
-#     # netcdf only
-#     vector_model.write_vector(fn="vector/vector_full.nc", fn_geom=None)
-#     # geojson only
-#     # automatic split
-#     vector_model.write_vector(fn=None, fn_geom="vector/vector_split.geojson")
-#     assert isfile(join(vector_model.root.path, "vector", "vector_split.nc"))
-#     assert not isfile(join(vector_model.root.path, "vector", "vector_all.nc"))
-#     # geojson 1D data only
-#     vector_model._vector = vector_model._vector.drop_vars("zs").drop_vars("time")
-#     vector_model.write_vector(fn=None, fn_geom="vector/vector_all2.geojson")
-#     assert not isfile(join(vector_model.root.path, "vector", "vector_all2.nc"))
-
-#     # test read vector
-#     vector_model1 = VectorModel(str(tmpdir), mode="r")
-#     # netcdf only
-#     vector_model1.read_vector(fn="vector/vector_full.nc", fn_geom=None)
-#     vector0 = vector_model1.vector
-#     assert len(vector0["zs"].dims) == 2
-#     vector_model1._vector = None
-#     # geojson only
-#     # automatic split
-#     vector_model1.read_vector(
-#         fn="vector/vector_split.nc", fn_geom="vector/vector_split.geojson"
-#     )
-#     vector1 = vector_model1.vector
-#     assert len(vector1["zs"].dims) == 2
-#     vector_model1._vector = None
-#     # geojson 1D data only
-#     vector_model1.read_vector(fn=None, fn_geom="vector/vector_all2.geojson")
-#     vector3 = vector_model1.vector
-#     assert "zs" not in vector3
+def test_vectormodel(vector_model, tmpdir):
+    # write model
+    vector_model.root.set(str(tmpdir), mode="w")
+    vector_model.write()
+    # read model
+    model1 = Model(
+        root=str(tmpdir),
+        mode="r",
+        components={
+            "vector": {"type": VectorComponent.__name__},
+            "config": {"type": ConfigComponent.__name__},
+        },
+    )
+    model1.read()
+    equal, errors = vector_model.test_equal(model1)
+    assert equal, errors
+
+
+def test_vectormodel_vector(vector_model_no_defaults, tmpdir, geoda):
+    # test set vector
+    testds = vector_model_no_defaults.vector.data.copy()
+    vector_component = vector_model_no_defaults.get_component("vector", VectorComponent)
+    # np.ndarray
```

Comment:
I think it would be nice to subdivide these into multiple tests, but you can also leave that for later if you want, as it's not critical and this PR is already quite large.

Reply:
not now :)